src/OrganisationRegistry.Api/Security/SecurityService.cs

```diff
@@ -228,16 +212,66 @@ public bool CanUseLabelType(IUser user, Guid labelTypeId)
             return true;
         }
 
+        private static bool HasPermissionsForOrganisation(SecurityInformation securityInfo, Guid? organisationId)
+        {
+            if (!organisationId.HasValue)
+                return false;
+
+            return
+                securityInfo.Roles.Contains(Role.OrganisatieBeheerder) &&
+                securityInfo.OrganisationIds.Contains(organisationId.Value);
+        }
+
+        private static bool HasPermissionsForBody(SecurityInformation securityInfo, Guid? bodyId)
+        {
+            if (!bodyId.HasValue)
+                return false;
+
+            return
+                securityInfo.Roles.Contains(Role.OrganisatieBeheerder) &&
+                securityInfo.BodyIds.Contains(bodyId.Value);
+        }
+
+        private async Task<OrganisationSecurityInformation> GetOrganisationSecurityInformation(
+            ClaimsPrincipal user,
+            bool isAuthenticated,
+            string subject)
+        {
+            var maybeCachedSecurityInfo = isAuthenticated ? _cache.Get(subject) : null;
+            if (isAuthenticated && maybeCachedSecurityInfo is OrganisationSecurityInformation cachedSecurityInfo)
+            {
+                return cachedSecurityInfo;
+            }
+
+            if (isAuthenticated)
+            {
+                var organisations = GetOrganisations(user);
+                var organisationSecurity = await GetSecurityInformation(organisations);
+                _cache.Set(
+                    new CacheItem(subject, organisationSecurity),
+                    new CacheItemPolicy
+                    {
+                        SlidingExpiration = TimeSpan.FromMinutes(1)
```

Comment:
hmmm, there exists a config setting for sliding expiration ... shouldn't this code use that setting ? or maybe create another setting ?

Reply:
Yes!